core/runtime/src/main/java/io/quarkiverse/langchain4j/RegisterAiService.java

```diff
@@ -0,0 +1,156 @@
+package io.quarkiverse.langchain4j;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.function.Supplier;
+
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.ChatMemoryProvider;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.moderation.ModerationModel;
+import dev.langchain4j.retriever.Retriever;
+import dev.langchain4j.service.AiServices;
+
+/**
+ * Used to create Langchain4j's {@link AiServices} in a declarative manner that the application can then use simply by
+ * using the class as a bean.
+ * Under the hood Langchain4j's {@link AiServices#builder(Class)} is called
+ * while also providing the builder with the proper {@link ChatLanguageModel}, {@code tools} beans,
+ * {@link ChatMemory} or {@link ChatMemoryProvider}, {@link ModerationModel} and {@link Retriever}
+ */
+@Retention(RUNTIME)
+@Target(ElementType.TYPE)
+public @interface RegisterAiService {
+
+    /**
+     * Configures the way to obtain the {@link ChatLanguageModel} to use.
+     * If not configured, the CDI bean implementing the model is looked up.
+     * Such a bean provided automatically by extensions such {@code quarkus-langchain4j-openai} and
+     * {@code quarkus-langchain4j-hugging-face}
+     */
+    Class<? extends Supplier<ChatLanguageModel>> chatLanguageModelSupplier() default BeanChatLanguageModelSupplier.class;
+
+    /**
+     * Tool classes to use. All tools are expected to be CDI beans.
+     * <p>
+     * NOTE: when this is used, either {@code chatMemorySupplier} or {@code chatMemoryProviderSupplier} MUST be set.
+     */
+    Class<?>[] tools() default {};
```

Comment:
I'm wondering about the default. Should it be "all tools" instead of none?

Reply:
You mean all tools we find on the classpath?

Reply:
I think that would be a little too susprising as a default.